lumicks/pylake/detail/image.py

```diff
@@ -114,10 +114,8 @@ def line_timestamps_image(time_stamps, infowave, pixels_per_line):
     pixels_per_line : int
         The number of pixels on the fast axis of the scan.
     """
-    pixel_start_idx = np.flatnonzero(infowave == InfowaveCode.pixel_boundary)
-    pixel_start_idx = np.concatenate(([0], pixel_start_idx + 1))
-
-    return time_stamps[pixel_start_idx[0:-1:pixels_per_line]]
+    min_stamps = reconstruct_image(time_stamps, infowave, shape=(pixels_per_line,), reduce=np.min)
+    return min_stamps.astype(np.int64)[:, 0]
```

Comment:
I'm not sure how often this would come up, but `reconstruct_image()` is a relatively expensive calculation; so if a user creates many slices from a single parent kymograph this calculation would be repeated many times.

With this change, the code is very similar to the `ConfocalImage._timestamps()` method which is cached. Would it make sense to instead make this function a cached method of `Kymo`? As far as I can tell, `Kymo.__getitem__()` is the only place where `line_timestamps_image()` is used.

Reply:
Yeah, I considered that.

When the kymo is sliced a new instance is created, so it wouldn't benefit from the caching when doing repeated slicing. I guess you would benefit if you'd take multiple slices from the same kymo though.

I can make it a cached private member function.

Reply:
Yes, my concern is if you make like 5 slices from the same Kymo you repeat that calculation 5 times

Reply:
Yeah, it's a good point, should be fixed now!